### PR TITLE
V0.5.1

### PR DESCRIPTION
**Pull request by mountainMath**
Placeholder for next release. Still needs to incorporate https://github.com/mountainMath/cancensus/issues/181, https://github.com/mountainMath/cancensus/issues/180, https://github.com/mountainMath/cancensus/issues/179, and https://github.com/mountainMath/cancensus/issues/182.

### CODE REVIEW

**Comment by dshkol** [comment]
What is geo_suite meant to do?

**Comment by mountainMath** [comment]
It’s supposed to access GeoSuite data. That’s basically a comprehensive file for each geographic level with all higher level geographic identifiers, and also population, household and dwelling counts.
It’s useful for traversing geographic hierarchies and much easier and faster than downloading data bit by bit from CensusMapper.

**Review by dshkol — COMMENTED** [review]
thinking about backwards compatibility here, we should probably let the deprecated function call still work for now or fail gracefully, otherwise might cause some issues in existing workflows.

**Comment by mountainMath** [comment]
GeoSuite wasn't part of any prior release, I added this in for the current release. But there are several issues that have to be worked out before it's ready.
1. The R unzip functions can't handle it, or at least not in a cross-platform way because of special characters in file names.
2. The data is inconsistently organized across census years.

Maybe one way around this is that I first do some pre-processing and then dump them as files on CensusMapper or in an S3 bucket and the package accesses the cleaned data from there.

**Comment by mountainMath** [comment]
I think everything is implemented, except for [the install checks for {sf}](https://github.com/mountainMath/cancensus/issues/180). Should we  bump that to the next version @dshkol?

**Comment by dshkol** [comment]
Noticed that `geo_suite.R` file is still present, and `cancensus:::get_geo_suite()` still exists in the namespace. Is that intended?

File: README.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ For larger quotas, please get in touch with Jens [directly](mailto:jens@censusma
...
 
+Starting with version 0.5.1 **cancensus** will automatically check if locally cached data has been recalled by StatCan via the new data recall API implemented in [CensusMapper](https::/censusmapper.ca). StatCan occasionally detects and corrects errors in their census data release, and **cancensus** will download a list of recalled data at the start of each session and emit a warning if it detected locally cached data that has been recalled. Removal of the cached recalled data has to be done explicitly by the user via the `remove_recalled_chached_data()` function. If data was cached with **cancenus** versions prior to version 0.5.0 there is insufficient metadata to determine all instances of recalled cached data, but the package will check every time cached data is loaded and can identify recalled data at this point at the latest and issues a warning if recalled data is loaded.
```
Comment (dshkol) [inline README.md:50]: > ... will download a list of recalled data at the start of each session and emit a warning if it detected locally cached data that has been recalled.

What if users are offline and are working with saved or cached data? I am not sure a silent API check and download is something _all_ users would agree to.

File: R/cancensus.R [hunk 2]
```diff
@@ -12,10 +12,10 @@
...
 #' @param geo_format By default is set to \code{NA} and appends no geographic information. To include geographic information with census data, specify one of either \code{"sf"} to return an \code{\link[sf]{sf}} object (requires the \code{sf} package) or \code{"sp"} to return a \code{\link[sp]{SpatialPolygonsDataFrame-class}} object (requires the \code{rgdal} package).
-#' @param resolution Resolution of the geographic data. {cancensus} will download simplified geometries by default. For lower level geometries like DB or DA this will be very close to the high resoltion data,
-#' siplification level increases as the geographic aggregation level increases.
+#' @param resolution Resolution of the geographic data. {cancensus} will download simplified geometries by default. For lower level geometries like DB or DA this will be very close to the high resolution data.
+#' Simplification generally increases as the geographic aggregation level increases.
 #' If high resolution geometries are required
-#' then this option can be set to 'high'. By default is set to \code{'simplified'}.
+#' then this option can be set to 'high'. By default this setting is set to \code{'simplified'}.
```
Comment (dshkol) [inline R/cancensus.R:18]: Suggest "max"/"simplified" as an alternative to "high"/"simplified"

**Comment by dshkol** [comment]
> I think everything is implemented, except for [the install checks for {sf}](https://github.com/mountainMath/cancensus/issues/180). Should we bump that to the next version @dshkol?

I can still add that but am swamped with work this week. If you want to try and push this version to CRAN, we can defer it til next time and focus on that in the next couple days pending my questions around automatically checking recall status.

File: cran-comments.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # Update - v0.5.1
-- functionality to detect and remove locally cached data that has been recalled
-- automatic checks that warn if locally cached data has been recalled
-- rename get/set cache/api_key functions to avoid name conflicts with other packages
+- Added functionality to detect and remove locally cached data that has been recalled by Statistics Canada
+- added automatic checks that warn if locally cached data has been recalled
```
Comment (dshkol) [inline cran-comments.md:3]: concerned this might be an issue

File: R/cancensus.R (hunk 2 above)
Comment (mountainMath) [reply R/cancensus.R:18]: I'd rather keep "high", all geographies have been processed. "max" implies that they have maximal resolution, but all geographies are simplified to some degree and they are also heavily processed in the sense that water features (simplified depending to geographic level) have been removed.

File: README.md (hunk 1 above)
Comment (mountainMath) [reply README.md]: The download happens on the first invocation of get_census, will clarify that. If users are offline this fails gracefully, just emitting a warning message that recall data isn't available on the first run, and everything runs through normally.

**Comment by mountainMath** [comment]
The geo_suite file is still part of the repo but added to the .Rbuildignore, will check into why it's part of the namespace. Left the file in there as a working stub, but it seems it still shows up in the namespace and that's an issue. Will experiment how to get rid of it completely, might just move it into a separate branch.

File: cran-comments.md (hunk 3 above)
Comment (mountainMath) [reply cran-comments.md:3]: I think the automatic checks are good, users don't read the package notes or re-read the README/documentation when updating packages. Warning users that they are using recalled data is important.

File: R/cancensus.R (hunk 2 above)
Comment (dshkol) [reply R/cancensus.R:18]: sounds good, but I wonder if at some point we should be more transparent that the spatial data people are using is (1) not the StatCan master original and (2) simplified and therefore not suitable for anything other than simple thematic mapping and analysis. I am sure no body is making planning decisions using these geometries and maybe I'm just being a bit pedantic.

File: cran-comments.md (hunk 3 above)
Comment (dshkol) [reply cran-comments.md:3]: Yeah, just needed to rephrase that this is part of a user's initial call and is activated by that only, not something that happens in the background without the user knowing

File: README.md [hunk 4]
```diff
@@ -122,9 +122,9 @@ There are several other jurisdiction where census data is available via R packag
 * Chile: [chilemapas](https://pacha.dev/chilemapas/)
 * Czech Republic: [RCzechia](https://github.com/jlacko/RCzechia)
 * Finland: [geofi](https://ropengov.github.io/geofi/)
-* Ghana: [rGhanaCensus](https://cran.r-project.org/web/packages/rGhanaCensus/index.html)
+* Ghana: [rGhanaCensus](https://CRAN.R-project.org/package=rGhanaCensus)
```
Comment (dshkol) [inline README.md:125]: All CRAN package links need to take the "canonical" form like this

File: README.md [hunk 5]
```diff
@@ -122,9 +122,9 @@ There are several other jurisdiction where census data is available via R packag
...
 * Spain: [mapSpain](https://github.com/rOpenSpain/mapSpain/)
-* UK: [geographr](https://github.com/britishredcrosssociety/geographr)
+* UK: [geographr](https://github.com/humaniverse/geographr)
```
Comment (dshkol) [inline README.md:127]: Moved - picked up in cran pretesting

**Pull request merged**